src/compressed_tensors/quantization/observers/base.py

```diff
@@ -95,13 +99,26 @@ def get_qparams(
                 scales, zero_points = [], []
                 group_idxs = range(0, columns, self.quantization_args.group_size)
                 for group_id, group_idx in enumerate(group_idxs):
-                    scale, zero_point = self.get_qparams_along_dim(
-                        observed[:, group_idx : (group_idx + group_size)],
-                        0,
-                        tensor_id=group_id,
-                    )
-                    scales.append(scale)
-                    zero_points.append(zero_point)
+
+                    # initialized g_idx are Tensor of -1s
+                    if g_idx is not None and group_idx // group_size in g_idx:
```

Comment:
we might want to move this to a helper function to share between forward and observers/base

something like `iter_weight_by_groups(tensor, g_idx: Optional = None) -> Generator`